src/main/java/zd/zero/waifu/motivator/plugin/WaifuMotivatorProject.java

```diff
@@ -88,6 +88,11 @@ private void checkIfInGoodState( Project projectOpened, Runnable onGoodState ) {
     public void projectClosing( @NotNull Project project ) {
         if ( project == this.project ) dispose();
 
+        if ( !getPluginState().isOnboardingFinished() ) {
```

Comment:
This will fix the issue for our user's first installation experience (I think, haven't tested it yet :)

However, this is a dynamic plugin now, so this issue would apply to any future compatible updates.

I was hoping to re-imagine the changes done here 

https://github.com/zd-zero/waifu-motivator-plugin/blob/ad13b4b06ac848a864d7f187da7a60299a1b4f5f/src/main/java/zd/zero/waifu/motivator/plugin/listeners/PluginInstallListener.kt#L26

I just sort of hack that together. I think it might be useful to looking into registering listeners to each project if they are not already (on install)

What do you think?

Reply:
It's either that or setting the installation semaphore in the plugin install listener, and update it after the project has reloaded.

Reply:
Hmmmm, the condition only applies once for the install, I can't seem to understand the motivation behind here. If possible, can you give more problem statements or elaborate more on the scenario so I can understand it well 🙏.

Reply:
Sure. 

So when the plugin is first installed this class is loaded, and it will see that it was installed because of this method: https://github.com/zd-zero/waifu-motivator-plugin/blob/ad13b4b06ac848a864d7f187da7a60299a1b4f5f/src/main/java/zd/zero/waifu/motivator/plugin/listeners/PluginInstallListener.kt#L23

This also applies to updating the plugin. So the plugin is already installed and the user is taking the new update. The plugin will see that it was loaded and trigger the same logic.

In the end, first installation and any updates to after will trigger all projects to be reloaded (as the code sits).

So I think it would be helpful to also address the updates to the plugins as well. As this is currently just targeting the first installs.

Reply:
Unless I missed something.

Reply:
I think I get it now, do you intend to have it checked dynamically instead of adding a new property state?

Reply:
Yep. So the dynamic plugin listener is the one causing both of these issues, so you could potentially fix both issues by utilizing the plugin state. I added the listener because of this issue #198 and now it is being a jerk.